packages/ag-charts-community/src/chart/series/series.ts

```diff
@@ -569,7 +568,16 @@ export abstract class Series<
         return direction;
     }
 
-    abstract getDomain(direction: ChartAxisDirection): any[];
+    // The union of the series domain ('community') and series-option domains ('enterprise').
+    getDomain(direction: ChartAxisDirection): any[] {
+        const seriesDomain: any[] = this.getSeriesDomain(direction);
+        const moduleDomains: any[][] = this.dispatch('data-getDomain', { direction }) ?? [];
+        // Flatten the 2D moduleDomains into a 1D array and concatenate it with seriesDomain
+        return moduleDomains.reduce((total, current) => total.concat(current), seriesDomain);
+    }
+
+    // Get the 'community' domain (excluding any additional data from series-option modules).
+    abstract getSeriesDomain(direction: ChartAxisDirection): any[];
```

Comment:
Maybe we should name this `getDataDomain()`, as the `Series` seems redundant otherwise in the context of a method on `Series`?

Reply:
I somewhat agree about the redundancy.

However the name `getDataDomain()` is a bit too vague; the lower/upper error bounds are also part of the input data, so this method name could be misunderstood to mean "domain of the data in the series and the series-options modules".

I've deliberately named this `getSeriesDomain()` to clarify that this is the domain of the series excluding any extra data points from series-options modules.

I initially used the name `getLocalDomain()` (series) and `getTotalDomain()` (series + series-options), but the wording `Local` felt too abstract.

Please let me know your thoughts.

Reply:
Ok - let's roll with it then 👍